### ISSUE TITLE
Supported MPI libraries

### ISSUE THREAD

**Issue by ivan-pi** [jeffhammond/vapaa#41]
I noticed there is a version constraint `MPICH_NUMVERSION > 40200000`. Is this a hard constraint?

This rules out Intel MPI which exports the constant with a value of `30400002`.

**Comment by jeffhammond** [comment]
I'm using an MPICH extension for noncontiguous subarrays that requires that version. If you don't have it, I don't think everything works, but what doesn't work isn't used by anybody, so it's safe.

**Comment by jeffhammond** [comment]
Yeah I can share slides. I'll upload here tomorrow.

**Comment by jeffhammond** [comment]
[Hammond MPI Fortran PASC24 post.pdf](https://github.com/user-attachments/files/15814652/Hammond.MPI.Fortran.PASC24.post.pdf)
@ivan-pi slides here

**Comment by jeffhammond** [comment]
> So if I get rid of the procedures that uses [`MPIX_Type_iov`](https://www.mpich.org/static/docs/v4.1.x/www3/MPIX_Type_iov.html), it should work in principle?

Yes, you can add some preprocessor stuff and it will work for all MPI, but if you do unreasonable things that require MPICH 4+, then it will fail.
